File: app/src/main/java/com/dream/countryinfo/network/CountryApi.java
package com.dream.countryinfo.network;

import com.dream.countryinfo.feature.country.CountryDetail;

import java.util.List;
import java.util.Map;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;
import rx.Observable;

/**
 * Created by lixingming on 09/11/2017.
 */

public interface CountryApi {

    @GET("rest/v2/name/{name}?fullText=true")
    Call<List<CountryDetail>> getCountriesByName(@Path("name") String name, @Query("fields") String fields);

    @GET("rest/v2/name/{name}")
    Call<List<Map<String, String>>> searchCountriesByName(@Path("name") String name, @Query("fields") String fields);

    @GET("rest/v2/all")
    Call<List<Map<String, String>>> getAllCountries(@Query("fields") String fields);

    @GET("rest/v2/name/{name}")
    Observable<String> getOneCountry(@Path("name") String name, @Query("fields") String fields);

}

File: app/src/main/java/com/dream/countryinfo/network/CountryApi.kt
package com.dream.countryinfo.network

import com.dream.countryinfo.feature.country.CountryDetail

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

/**
 * Created by lixingming on 09/11/2017.
 */

interface CountryApi {

    @GET("rest/v2/name/{name}?fullText=true")
    fun getCountriesByName(@Path("name") name: String, @Query("fields") fields: String): Call<List<CountryDetail>>

    @GET("rest/v2/name/{name}")
    fun searchCountriesByName(@Path("name") name: String, @Query("fields") fields: String): Call<List<Map<String, String>>>

    @GET("rest/v2/all")
    fun getAllCountries(@Query("fields") fields: String): Call<List<Map<String, String>>>

}

